sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -260,8 +271,19 @@ def beautify_dom_threshold_info_dict(dom_info_dict):
     dom_info_dict['txbiashighwarning'] = strip_unit_and_beautify(dom_info_dict['txbiashighwarning'], BIAS_UNIT)
     dom_info_dict['txbiaslowwarning'] = strip_unit_and_beautify(dom_info_dict['txbiaslowwarning'], BIAS_UNIT)
 
-# Update port sfp info in db
+# Reset/re-initialize the transciever to its default state
 
+def reset_transceiver(logical_port_name):
+    physical_port_list = logical_port_name_to_physical_port_list(logical_port_name)
+    if physical_port_list is None:
+        helper_logger.log_error("No physical ports found for logical port '{}'".format(logical_port_name))
+        return
+
+    for physical_port in physical_port_list:
```

Comment:
Module reset is not required upon module insertion. Can we use sleep instead?.

Reply:
can you check CMIS spec : http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf (see D.1.3 Software Configuration and Initialization)

Reply:
Also, this logic would introduce 8 seconds sleep for a single 4x100G.
For e.g. If Ethernet0,2,4,6 is part of single physical port(1), delay will be executed for each port(0/2/4/6).

Reply:
yes, you are right

Reply:
On top of this, the insertion events are processed serially, so even if there was only one sleep done per physical port on a 32-port system this would result in over a minute of delay at startup, even though the ports were all "inserted" at the same time, right?

Reply:
@prgeor , can you check the boot up sequence, will this introduce cumulative delay?

Reply:
yes, if there are 32 optics populated in a 32-ports system then the total delay will be 32 * 2 secs = 64 secs.

Reply:
can you then fix this properly?

Reply:
reset_transceiver() is already removed

Reply:
Removed the cumulative delay across port insertions